File: 1189. Maximum Number of Balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {

        unordered_map<char, int> freq;

        // Count each character
        for (char ch : text) {
            freq[ch]++;
        }

        // Find maximum number of "balloon"
        return min({
            freq['b'],
            freq['a'],
            freq['l'] / 2,
            freq['o'] / 2,
            freq['n']
        });
    }
};

File: 1189. Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:

        freq = {}

        # Count each character
        for ch in text:
            freq[ch] = freq.get(ch, 0) + 1

        # Find maximum number of "balloon"
        return min(
            freq.get('b', 0),
            freq.get('a', 0),
            freq.get('l', 0) // 2,
            freq.get('o', 0) // 2,
            freq.get('n', 0)
        )
